### PR TITLE
Handle bad external data better

### PR DESCRIPTION
**Pull request by GlazerMann**
Fixes # .

Changes proposed in this pull request:
-
-
-

### CODE REVIEW

**Comment by GlazerMann** [comment]
This also makes sure that the two ZooKeys areas of code stay in sync

**Comment by GlazerMann** [comment]
This pull:
1.  Use constant arrays to store data for easy changing
2.  Add a test case
3.  For certain "bad" authors, like Time Inc, recognize them as publishers and use that data

File: Template.php [hunk 1]
```diff
@@ -22,6 +22,10 @@ class Template extends Item {
   const PLACEHOLDER_TEXT = '# # # Citation bot : template placeholder %s # # #';
   const REGEXP = '~\{\{(?:[^\{]|\{[^\{])+?\}\}~s';
   const TREAT_IDENTICAL_SEPARATELY = FALSE;
+  const BAD_AUTHORS = array("unknown"); // use lower case
+  const AUTHORS_ARE_PUBLISHERS = array("hearst magazines", "time inc"); // use lower case
+  const HAS_NO_VOLUME = array("zookeys"); // use lower case
+  const BAD_TITLES = array("unknown");  // use lower case
```
Comment (ms609) [inline Template.php]: Can constants that are not specific to the Template class be defined in constants.php, please?

**Comment by GlazerMann** [comment]
Your requested changed have been made.

**Pull request merged**